### ISSUE TITLE
hough_line_peaks misses peaks.

### ISSUE THREAD

**Issue by hohlraum** [scikit-image/scikit-image#1420]
Below is an example of ```label_hspace``` where the peak below 1 is missed by the peak finder. The problem seems to be that the masked ```hspace``` has no intensity at the centroid. It seems to work if L87 is updated to use ```hspace_max``` instead of the masked ```hspace```.
```
        accum = hspace_max[dist_idx, angle_idx]
```
https://github.com/scikit-image/scikit-image/blob/master/skimage/transform/hough_transform.py#L87

![download](https://cloud.githubusercontent.com/assets/3930880/6610530/da381f3c-c860-11e4-8a21-6adbb3a52398.png)

**Comment by hohlraum** [comment]
Looks like there's an even bigger problem. Peak candidates are found with ```label```, and coordinates generated with ```measure.regionprops```. Currently this is a list, sorted leftmost peaks first. Peak candidates are assessed in this order so an early peak will suppress its neighbours even if the later peaks have higher intensities. This is particularly noticeable when low value peaks near -pi/2 suppress higher valued ones near +pi/2.

The solution is to sort the ```props``` list based on the maximum intensity of each label.

**Comment by stefanv** [comment]
That's a good catch @hohlraum.  Are you willing to help us patch up this bug?

**Comment by stefanv** [comment]
FWIW, I am not very happy with our state of peak detectors in skimage--another area that may require attention.  But I think we can solve this problem separately.

**Comment by hohlraum** [comment]
The PR includes an intensity sort that addresses the second issue.

There's another problem with the max filter applied at the top of the function. The exclusion neighbourhood values ```min_distance``` and ```min_angle``` are being used as the window for the 1d max filters. The purpose of this filter (I believe) is to expand the peak pixels for the subsequent label. It is not meant to define the suppression windows, that happens later.

Consider what happens when I have three peaks of intensities 1,2,3 equally spaced by 10 units. If I set ``` min_distance=11``` I expect to receive peaks 1 and 3 since 3 suppresses 2, but not 1. However, because of the max filter 2 ends up suppressing 1 because the it fails the mask test. Peak 1 never even shows up in the ```coords``` list.

It seems to me that ```distance_size``` and ```angle_size``` should not be dependant on the values for ```min_distance``` and ```min_angle```. To fix it I've just hardcoded them to something small. I'm not sure whether this is the correct solution. Would it be better to have two more arguments?

Finally, using max has the effect of not preserving integrated intensity. Two nearby, equally spaced peaks will be maxed to a single peak having the same intensity, not the sum. It seems to me that each labelled region should have the integrated intensity from the area covered by the label. 

I'll see if I can bash together something better.
